tx-indexer/src/collector.rs

```diff
@@ -21,14 +19,20 @@ const PROBLEMATIC_BLOCKS: [near_indexer_primitives::CryptoHash; 2] = [
 
 #[cfg_attr(feature = "tracing-instrumentation", tracing::instrument(skip_all))]
 pub(crate) async fn index_transactions(
-    chain_id: config::ChainId,
     streamer_message: &near_indexer_primitives::StreamerMessage,
     db_manager: &std::sync::Arc<Box<dyn database::TxIndexerDbManager + Sync + Send + 'static>>,
     tx_collecting_storage: &std::sync::Arc<impl TxCollectingStorage>,
+    indexer_config: &configuration::Config,
 ) -> anyhow::Result<()> {
-    extract_transactions_to_collect(streamer_message, db_manager, tx_collecting_storage).await?;
+    extract_transactions_to_collect(
+        streamer_message,
+        db_manager,
+        tx_collecting_storage,
+        indexer_config,
+    )
+    .await?;
     collect_receipts_and_outcomes(
-        chain_id,
+        indexer_config.general.chain_id.clone(),
```

Comment:
Could the `collect_receipts_and_outcomes` accept a reference for the chain ID to avoid cloning? (not a hard requirement though)